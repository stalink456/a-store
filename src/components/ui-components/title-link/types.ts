import { TitleProps } from '@alfalab/core-components/typography';

export type TitleLinkType = TitleProps & {
  text: string;
  route: string;
  children?: React.ReactNode;
};
