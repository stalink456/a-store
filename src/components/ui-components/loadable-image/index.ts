export { LoadableImage } from './loadable-image';
