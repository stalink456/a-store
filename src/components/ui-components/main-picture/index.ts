export { MainPicture } from "./main-picture";
