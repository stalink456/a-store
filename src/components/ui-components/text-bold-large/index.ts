export { TextBoldLarge } from "./text-bold-large";
