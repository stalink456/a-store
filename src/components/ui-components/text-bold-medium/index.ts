export { TextBoldMedium } from "./text-bold-medium";
