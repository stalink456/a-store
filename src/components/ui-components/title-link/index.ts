export { TitleLink } from './title-link';
