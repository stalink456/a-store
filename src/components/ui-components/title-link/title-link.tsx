import React from 'react';
import { Link } from 'react-router-dom';
import { Typography } from '@alfalab/core-components/typography';
import { TitleLinkType } from './types';

export const TitleLink: React.FC<TitleLinkType> = React.memo(
  ({ view, tag, weight, text, route, children }) => {
    return (
      <React.Fragment>
        <Link to={route}>
          <Typography.Title tag={tag} weight={weight} view={view}>
            {text}
          </Typography.Title>
          {children}
        </Link>
      </React.Fragment>
    );
  }
);
