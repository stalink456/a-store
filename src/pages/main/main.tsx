import React from 'react';
import { MainPicture } from 'components/ui-components/main-picture';
import madeInAlfa from 'images/MadeInAlfa.jpeg';
import yourDesign from 'images/YourDesign.jpeg';

import styles from './main.module.css';

export const Main: React.FC = () => {
  return (
    <div className={styles.main}>
      <MainPicture
        src={madeInAlfa}
        text='Сделано в Альфе'
        route='/made-in-alfa'
      />
      <MainPicture src={yourDesign} text='Свой дизайн' route='/your-design' />
    </div>
  );
};
