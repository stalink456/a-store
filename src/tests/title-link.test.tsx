import { MemoryRouter, Route, Routes } from "react-router-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { TitleLink } from "components/ui-components/title-link";
import { Main } from "pages/main";

describe("TitleLink component", () => {
  it("should render TitleLink component", () => {
    const { baseElement } = render(
      <TitleLink tag="h1" text="A-Store" route="/" />,
      { wrapper: MemoryRouter }
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should render TitleLink component with title", async () => {
    render(<TitleLink tag="h1" text="A-Store" route="/" />, {
      wrapper: MemoryRouter,
    });

    expect(await screen.findByText("A-Store")).toBeInTheDocument();
  });

  it("should render TitleLink component with a transition to Main page", async () => {
    const route = "/your-design";

    render(
      <MemoryRouter initialEntries={[route]}>
        <Routes>
          <Route
            path="/your-design"
            element={<TitleLink tag="h1" text="A-Store" route="/" />}
          />
          <Route path="/" element={<Main />} />
        </Routes>
      </MemoryRouter>
    );

    fireEvent.click(await screen.findByText("A-Store"));

    expect(await screen.findByText("Свой дизайн")).toBeInTheDocument();
  });

  it("shouldn't render TitleLink component with non-existent text", async () => {
    render(<TitleLink tag="h1" text="A-Store" route="/" />, {
      wrapper: MemoryRouter,
    });

    expect(screen.queryByText("non-existent text")).not.toBeInTheDocument();
  });
});
