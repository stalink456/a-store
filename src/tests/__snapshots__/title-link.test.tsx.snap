// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleLink component should render TitleLink component 1`] = `
<body>
  <div>
    <a
      href="/"
    >
      <h1
        class="typography__component_19zgr typography__styrene-medium_pxbut typography__medium_pxbut"
      >
        A-Store
      </h1>
    </a>
  </div>
</body>
`;
